chore: clarify user context UI and interface (#13620)
getsentry/sentry-docs: develop-docs/sdk/data-model/event-payloads/user.mdx

affected by one issue, for example. Sending a user that has none
of these attributes and only custom attributes is valid, but not as useful.

<Alert title="Note" level="warning">

> These default keys are case-sensitive!

Author: ```suggestion
These keys are case-sensitive, so mistyping will lead to them being processed like custom attributes.
```

> These keys are case-sensitive, so mistyping will lead to them being processed like custom attributes.

Reviewer: ```suggestion
These keys are case-sensitive, so any mistyped keys will be processed as custom attributes.
```

All other keys are stored as extra information but not specifically processed by
Sentry.

<Alert title="Note">

> Titles of these well-known attributes are prettified in the UI. For example, `ip_address` is shown as `IP Address`. Other keys are shown as they were sent, with the prefix `data.`.

Reviewer: ```suggestion
The titles of these well-known attributes are prettified in the UI. For example, `ip_address` is shown as `IP Address`. Other keys are shown as they were sent, with the prefix `data.`.
```

> The listed attribute titles are prettified in the UI. For example, `ip_address` is shown as `IP Address`. Other keys are shown as they were sent, with the prefix `data.`. This also holds for mistyped keys, for example sending `Username` instead of `username` will show up as `data.Username`.

Reviewer 2: ```suggestion
Titles of these well-known attributes are prettified in the UI. For example, `ip_address` is shown as `IP Address`. Other keys are shown as they were sent, with the prefix `data.`.
```
